components/Ripple/src/private/MDCRippleLayer.m

```diff
@@ -28,26 +28,23 @@
 static NSString *const kRippleLayerPositionString = @"position";
 static NSString *const kRippleLayerScaleString = @"transform.scale";
 
-@implementation MDCRippleLayer {
-  CGFloat _rippleRadius;
+static CGFloat DefaultRippleRadius(CGRect rect) {
+  return (CGFloat)(MDCHypot(CGRectGetMidX(rect), CGRectGetMidY(rect)) + kExpandRippleBeyondSurface);
 }
 
+@implementation MDCRippleLayer
+
 - (void)setNeedsLayout {
   [super setNeedsLayout];
-  [self setRadiiWithRect:self.bounds];
+
   [self setPathFromRadii];
   self.position = CGPointMake(CGRectGetMidX(self.bounds), CGRectGetMidY(self.bounds));
 }
 
-- (void)setRadiiWithRect:(CGRect)rect {
-  _rippleRadius =
-      (CGFloat)(MDCHypot(CGRectGetMidX(rect), CGRectGetMidY(rect)) + kExpandRippleBeyondSurface);
-}
-
 - (void)setPathFromRadii {
-  CGRect ovalRect =
-      CGRectMake(CGRectGetMidX(self.bounds) - _rippleRadius,
-                 CGRectGetMidY(self.bounds) - _rippleRadius, _rippleRadius * 2, _rippleRadius * 2);
+  CGFloat radius = self.rippleRadius > 0 ? self.rippleRadius : DefaultRippleRadius(self.bounds);
+  CGRect ovalRect = CGRectMake(CGRectGetMidX(self.bounds) - radius,
+                               CGRectGetMidY(self.bounds) - radius, radius * 2, radius * 2);
```

Comment:
It looks like rippleRadius can affect `self.path`, but changing the value of self.rippleRadius doesn't appear to result in any logical path that would cause `self.path` to be updated. Is this expected?

Reply:
Good catch! I missed the setter for rippleRadius. I added behavior to cover this instance and test in [f0a34b4](https://github.com/material-components/material-components-ios/pull/7350/commits/f0a34b4deb1f56cda2cc6c133a8729f26f73694b).

Reply:
Becuase `[self setPathFromRadii];` is called every time before the ripple animation starts, is there a reason why you would need it in the setter too?

The ripple layers themselves are usually transient (unless in selected or tap is held), if setting the ripple layer alters `self.path` live, then it would affect the existing ripple and cause unwanted behavior. Ideally, setting the rippleRadius will affect future animations and not the existing layer. Therefore I would probably remove this setter, unless you think I'm missing something.

Reply:
I missed the call to `startRippleAtPoint...` calling `setPathFromRadii`. I don't think a setter is needed. @jverkoey Does that make sense?

Reply:
Fair points @yarneo. @codeman7 let's document the points @yarneo brought up in the API's docs so that another engineer doesn't accidentally make this change in the future.

Specifically:

```
maxRippleRadius will only affect new ripples. If a ripple is already being animated when this property is changed,
that ripple's radius will not be affected.
```